### ISSUE TITLE
Problem with startOfMonth function

### ISSUE THREAD

**Issue by anjali-karemani** [date-fns/date-fns#3293]
There is an issue with the [startOfMonth function](https://date-fns.org/v2.29.3/docs/startOfMonth)

  # I have used two functions startOfMonth(() and endOfMonth() in my program. endOfMonth() gives me correct last day of month, where as startOfMonth() gives me previous day of last month. 
I have tried with different time settings as well and the issue seems same.

  ### 💻 Code demonstrating the problem

  

  ```js
  import { fn } from 'date-fns'

  fn()
  //=> const firstDayOfMonth = startOfMonth(new Date());
      const lastDayOfMonth = endOfMonth(new Date());
  

 🙁 2022-11-30T18:30:00.000Z
       2022-12-31T18:29:59.999Z

  <!-- Describe what you received -->

  ### 🙂 Expected behavior

  <!-- Describe what you expected to get -->

  ## Debug information

  - date-fns version: <!-- Fill in the date-fns version you use -->
  - Browser/Node.js version: <!-- Fill in the envrionment version (i.e. Chrome 94.0.4606.61) -->
  - Your timezone: <!-- Fill in your timezone name (i.e. Asia/Singapore) -->
  - Your current time: <!-- Fill in the time when did you tested the probmlem (i.e. 16:20) -->

**Comment by mandaputtra** [comment]
What is the version that you're using? Seems there is no problem for me.

<img width="660" alt="Screenshot 2022-12-22 at 23 44 43" src="https://user-images.githubusercontent.com/23342943/209183973-cadfa323-2de5-4403-9248-8b9a99627c71.png">

**Comment by mrruby** [comment]
Same issue here

**Comment by mrruby** [comment]
I found the issue, it's because of timezone when your machine runs, startOfMonth returns the proper date but, because you are in UTC−12:00 to UTC it subtract the hours of diffenece between where you are located and UTC.

The solution would be to eg
```startDate: Date = parseISO("2022-12-01T00:00:00Z")```

and then 

```
const monthDate = addMonths(startDate, 2);
const start = startOfMonth(monthDate);
const formatedStartDate = format(start, "yyyy-MM-dd"),
```

**Comment by stopsopa** [comment]
Boils down to the conclusion I've mada in this Issue: https://github.com/date-fns/date-fns/issues/3482#issuecomment-1658056974

Long story short: there should be two sets of methods for each operation UTC and locale:

startOfMonth & startOfUTCMonth 
and  
endOfMonth & endOfUTCMonth

**Comment by razzeee** [comment]
Yeah, we're also having problems with this, specifically `startOfMonth`, as it usually goes to the month before. At least on our dev systems, but then doesn't do that on test and prod.

So that causes a bunch of headaches, IMO separating these would be great. Or alternatively have an option on the functions.

**Comment by razzeee** [comment]
Looking into this again today, it seems like just using https://github.com/date-fns/utc is the correct approach, as most (all?) functions in date-fns support that type too
